activesupport/lib/active_support/error_reporter.rb

```diff
@@ -19,17 +19,24 @@ module ActiveSupport
   #
   # Both methods can be restricted to only handle a specific exception class
   #
-  #  maybe_tags = Rails.error.handle(Redis::BaseError) { redis.get("tags") }
+  #   maybe_tags = Rails.error.handle(Redis::BaseError) { redis.get("tags") }
   #
   # You can also pass some extra context information that may be used by the error subscribers:
   #
-  #  Rails.error.handle(context: { section: "admin" }) do
-  #    # ...
-  #  end
+  #   Rails.error.handle(context: { section: "admin" }) do
+  #     # ...
+  #   end
   #
   # Additionally a +severity+ can be passed along to communicate how important the error report is.
   # +severity+ can be one of +:error+, +:warning+ or +:info+. Handled errors default to the +:warning+
   # severity, and unhandled ones to +error+.
+  #
+  # Both `handle` and `record` pass through the return value from the block. In the special case of `handle` handling an
+  # error, a fallback value can be provided that will be returned:
+  #
+  #   user = Rails.error.handle(fallback: User.anonymous) do
```

Comment:
Looks like `fallback` is a new option for `Rails.error.handle`... what if implemented to support a callable to allow some kind of lazyness? - different than a concrete value.

```ruby
fallback = -> { Today.general_holiday? ? DefaultInterestRate.new : InterestRate.costly_default_calculation }
interest_rate = Rails.error.handle(fallback: fallback) do
   InterestRate.find_by!(params)
end
```

Reply:
It makes sense to me.

Reply:
Yep. This makes total sense. In fact the example provide her only works if `.anonymous` is a singleton and not a factory.

I'll own this change. I'll get a PR up.

Reply:
I'll have to decide what to do if the `fallback.call` raises though. Probably just let it raise up out of the `rescue`.